lib/smart_proxy_openscap/openscap_report_parser.rb

```diff
@@ -1,111 +1,43 @@
 # encoding=utf-8
-require 'openscap'
-require 'openscap/ds/arf'
-require 'openscap/xccdf/testresult'
-require 'openscap/xccdf/ruleresult'
-require 'openscap/xccdf/rule'
-require 'openscap/xccdf/fix'
-require 'openscap/xccdf/benchmark'
-require 'json'
-require 'iconv' if RUBY_VERSION.start_with? '1.8'
+require 'tempfile'
 
 module Proxy::OpenSCAP
   class Parse
-    def initialize(arf_data)
-      OpenSCAP.oscap_init
-      size         = arf_data.size
-      @arf_digest  = Digest::SHA256.hexdigest(arf_data)
-      @arf         = OpenSCAP::DS::Arf.new(:content => arf_data, :path => 'arf.xml.bz2', :length => size)
-      @test_result  = @arf.test_result
-
-      @results      = @test_result.rr
-      @sds          = @arf.report_request
-      bench_source = @sds.select_checklist!
-      @benchmark    = OpenSCAP::Xccdf::Benchmark.new(bench_source)
-      @items        = @benchmark.items
-    end
-
-    def cleanup
-      @test_result.destroy if @test_result
-      @benchmark.destroy if @benchmark
-      @sds.destroy if @sds
-      @arf.destroy if @arf
-      OpenSCAP.oscap_cleanup
-    end
-
-    def as_json
-      parse_report.to_json
-    end
-
-    private
-
-    def parse_report
-      report        = {}
-      report[:logs] = []
-      passed        = 0
-      failed        = 0
-      othered         = 0
-      @results.each do |rr_id, result|
-        next if result.result == 'notapplicable' || result.result == 'notselected'
-        # get rules and their results
-        rule_data = @items[rr_id]
-        report[:logs] << populate_result_data(rr_id, result.result, rule_data)
-        # create metrics for the results
-        case result.result
-          when 'pass', 'fixed'
-            passed += 1
-          when 'fail'
-            failed += 1
-          else
-            othered += 1
+    include ::Proxy::Log
+    include ::Proxy::Util
+
+    def initialize(cname, policy_id, date)
+      @cname = cname
+      @policy_id = policy_id
+      @date = date
+    end
+
+    def as_json(arf_data)
+      json = nil
+      begin
+        json = Tempfile.open(filename, "/var/tmp") do |in_file|
```

Comment:
after in-person discussion I agree that block are worse, sorry, could you please revert to the original version? thanks, next time feel free to share your opinion before changing the code or keep it in second commit so my crazy ideas do not cost you extra energy :-)